src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -267,21 +269,31 @@ object DepositHandler {
     val itemsToResolve = fetchItems diff fetchItemsInBagStore
     for {
       _ <- resolveFetchItems(bagitDir, itemsToResolve)
-      _ <- removeFromFetchTxt(bagitDir, itemsToResolve)
+      _ <- if(itemsToResolve.isEmpty) Success(()) else pruneFetchTxt(bagitDir, itemsToResolve)
       bag = getBagFromDir(bagitDir)
       validationResult = bag.verifyValid
       _ <- handleValidationResult(bag, validationResult, fetchItemsInBagStore)
     } yield ()
   }
 
-  private def removeFromFetchTxt(bagDir: File, items: Seq[FetchTxt.FilenameSizeUrl]): Try[Unit] =
-    getFetchTxt(bagDir).map(fetchTxt => Try {
-      items.foreach(fetchTxt.remove)
-      if (fetchTxt.isEmpty && !new File(bagDir, "fetch.txt").delete()) throw new IllegalStateException("Unable to remove empty fetch.txt")
-      getBagFromDir(bagDir).write(new FileSystemWriter(bagFactory), bagDir)
-    }.map(_ => ()))
-    .getOrElse(Success(()))
+  def pruneFetchTxt(bagDir: File, items: Seq[FetchTxt.FilenameSizeUrl]): Try[Unit] =
+    getBagFromDir2(bagDir).map {
+      case bag =>
+        bag.getFetchTxt.removeAll(items.asJava)
```

Comment:
You did not incorporate the possibility that `bag.getFetchTxt` is `null` (which it will, if you look into the code!)